kratos/constraints/slip_constraint.cpp

```diff
@@ -0,0 +1,146 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "constraints/slip_constraint.h"
+
+namespace Kratos
+{
+
+SlipConstraint::SlipConstraint(
+    IndexType Id,
+    Node<3>& rNode,
+    const VariableComponentType& rVarX,
+    const VariableComponentType& rVarY,
+    const VariableComponentType& rVarZ,
+    const Variable<array_1d<double,3> >& rNormalVar
+    ) : BaseType(Id),
+        mrNode(rNode),
+        mpNormalVariable(&rNormalVar)
+{
+    mRelationMatrix.resize(1,2,false);
+    mRelationMatrix.clear();
+
+    mConstantVector.resize(1,false);
+    mConstantVector.clear();
+
+    DofPointerVectorType all_dofs;
+    all_dofs.reserve(3);
+    all_dofs.push_back(rNode.pGetDof(rVarX));
+    all_dofs.push_back(rNode.pGetDof(rVarY));
+    all_dofs.push_back(rNode.pGetDof(rVarZ));
+
+    auto n = rNode.FastGetSolutionStepValue(rNormalVar);
+    KRATOS_DEBUG_ERROR_IF(norm_2(n) < std::numeric_limits<double>::epsilon()) << "The norm of the normal vector is zero or almost zero" << std::endl;
```

Comment:
```suggestion
    KRATOS_ERROR_IF(norm_2(n) < std::numeric_limits<double>::epsilon()) << "The norm of the normal vector is zero or almost zero" << std::endl;
```

this check is only done in the constructor